docs/guides/evm/understanding_the_mining_process.rst

```diff
@@ -0,0 +1,181 @@
+
+For the sake of simplicity let's forget about transactions for now and assume we would want to mine an empty block.
+
+As a refresher, he's where we left of as part of the `Building Chains Guide <building_chains>`_.
+
+::
+
+  from evm.db.backends.memory import MemoryDB
+  from evm.db.chain import ChainDB
+  from evm.chains.mainnet import MAINNET_GENESIS_HEADER
+
+  # start a fresh in-memory db
+  chaindb = ChainDB(MemoryDB())
+
+  # initialize a fresh chain
+  chain = chain_class.from_genesis_header(chaindb, MAINNET_GENESIS_HEADER)
+
+Since we decided to not add any transactions to our block let's just call :func:`~~evm.chains.base.Chain.mine_block` and see what happens.
+
+::
+
+  # initialize a fresh chain
+  chain = chain_class.from_genesis_header(chaindb, MAINNET_GENESIS_HEADER)
+
+  chain.mine_block()
+
+Aw, snap! We're running into an exception at :func:`~evm.consensus.pow.check_pow`. Apparently we are trying to add a block to the chain that doesn't qualify the Proof-of-Work (PoW) rules. The error tells us precisely that the ``mix_hash`` of our block does not match the expected value.
+
+.. code-block:: console
+
+  Traceback (most recent call last):
+    File "scripts/benchmark/run.py", line 111, in <module>
+      run()
+    File "scripts/benchmark/run.py", line 52, in run
+      block = chain.mine_block()  #**pow_args
+    File "/py-evm/evm/chains/base.py", line 545, in mine_block
+      self.validate_block(mined_block)
+    File "/py-evm/evm/chains/base.py", line 585, in validate_block
+      self.validate_seal(block.header)
+    File "/py-evm/evm/chains/base.py", line 622, in validate_seal
+      header.mix_hash, header.nonce, header.difficulty)
+    File "/py-evm/evm/consensus/pow.py", line 70, in check_pow
+      encode_hex(mining_output[b'mix digest']), encode_hex(mix_hash)))
+
+  evm.exceptions.ValidationError: mix hash mismatch;
+  0x7a76bbf0c8d0e683fafa2d7cab27f601e19f35e7ecad7e1abb064b6f8f08fe21 !=
+  0x0000000000000000000000000000000000000000000000000000000000000000
+
+Let's lookup how :func:`~evm.consensus.pow.check_pow` is implemented.
+
+::
+
+  def check_pow(block_number: int,
+                mining_hash: Hash32,
+                mix_hash: Hash32,
+                nonce: bytes,
+                difficulty: int) -> None:
+      validate_length(mix_hash, 32, title="Mix Hash")
+      validate_length(mining_hash, 32, title="Mining Hash")
+      validate_length(nonce, 8, title="POW Nonce")
+      cache = get_cache(block_number)
+      mining_output = hashimoto_light(
+          block_number, cache, mining_hash, big_endian_to_int(nonce))
+      if mining_output[b'mix digest'] != mix_hash:
+          raise ValidationError("mix hash mismatch; {0} != {1}".format(
+              encode_hex(mining_output[b'mix digest']), encode_hex(mix_hash)))
+      result = big_endian_to_int(mining_output[b'result'])
+      validate_lte(result, 2**256 // difficulty, title="POW Difficulty")
```

Comment:
Sphinx has a [*literal include* directive](https://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/ShowingCodeExamplesInSphinx.html#literalinclude-directive) that can be used to inline a section of an external file.  Might be a good way to include this source code without actualy copy/pasting it.